### DISCUSSION TITLE
Web Adapter startup process breaks other lambda layers that modify startup

### DISCUSSION THREAD

**Discussion by rposbo-selfridges** [awslabs/aws-lambda-web-adapter#412] [Q&A]
Hi,

We're using the [Dynatrace OneAgent lambda layer](https://docs.dynatrace.com/docs/setup-and-configuration/setup-on-cloud-platforms/amazon-web-services/amazon-web-services-integrations/aws-lambda-integration/aws-lambda-extension/deploy-oneagent-on-lambda-container-images) to integrate nextjs applications, as containers, hosted in aws lambdas with our observability platform.

Unfortunately, this has proven impossible as the web adapter seems to trigger the lambda directly instead of modifying the runtime startup (?) - this is a bit beyond me, so the feedback from Dynatrace tech team is below:

> * the aws lambda web adapter starts the Lambda runtime by itself - this is inconsistent with the AWS Lambda standards; we adhere to these standards by modifying the runtime startup to start using our agent instead.

Is this something that can be altered such that both the web adapter and the dynatrace one agent lambda layer can work side by side?

## Expected Functionality
An AWS lambda hosting a nextjs application, as a container, using both the aws lambda adapter _and_ the Dynatrace OneAgent lambda layer, results in 1) a nextjs application that can both serve web requests, *and* 2) lambda telemetry reported to Dynatrace without instrumentation within the application code itself

## Actual Functionality
An AWS lambda hosting a nextjs application, as a container, using both the aws lambda adapter _and_ the Dynatrace OneAgent lambda layer, results in only a nextjs application that can serve web requests

**Comment by bnusunny** [comment]
In this setup, you could treat the nextjs application as a standalone application running in a docker container. Could you try their instructions for App Runner and let me know the results?

I don't have access to Dynatrace. Otherwise, I would have tested it myself.
